api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1381,25 +1382,106 @@ public void getObject(String bucketName, String objectName, String fileName)
+    ObjectStat objectStat = statObject(sourceBucketName, sourceObjectName);
+    String etag = objectStat.etag();
+    Date lastModified = objectStat.createdTime();
+    long length = objectStat.length();
+
+    // If the conditions specified are satisfied, get and put the object in new location
+    if (copyConditions.isUnmodifiedSince(lastModified) && copyConditions.isModifiedAfter(lastModified)
+        && copyConditions.etagMatches(etag) && copyConditions.etagDoesNotMatch(etag)) {
+      InputStream objectByteStream = getObject(sourceBucketName, sourceObjectName);
+      putObject(bucketName, objectName, objectByteStream, length, "application/octet-stream");
+      objectByteStream.close();
+    }
+  }
+
+  /**
+   * Returns a String bucketName splitting the objectPath passed to this method.
+   */
+  private static String getBucketNameFromPath(String objectPath) {
+    return objectPath.split("/")[1];
+  }
+
+  /**
+   * Returns a String objectName by splitting the objectPath passed to this method.
+   */
+  private static String getObjectNameFromPath(String objectPath) {
+    return objectPath.split("/")[2];
+  }
 
   /**
    * Returns an presigned URL to download the object in the bucket with given expiry time.
    *
-   * </p><b>Example:</b><br>
-   * <pre>{@code String url = minioClient.presignedGetObject("my-bucketname", "my-objectname", 60 * 60 * 24);
-   * System.out.println(url); }</pre>
+   * </p>
+   * <b>Example:</b><br>
+   * 
+   * <pre>
+   * {
+   *   &#64;code
+   *   String url = minioClient.presignedGetObject("my-bucketname", "my-objectname", 60 * 60 * 24);
+   *   System.out.println(url);
+   * }
+   * </pre>
    *
-   * @param bucketName  Bucket name.
-   * @param objectName  Object name in the bucket.
-   * @param expires     Expiration time in seconds of presigned URL.
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name in the bucket.
+   * @param expires
+   *          Expiration time in seconds of presigned URL.
    *
    * @return string contains URL to download the object.
    *
-   * @throws InvalidBucketNameException   upon an invalid bucket name
-   * @throws InvalidKeyException          upon an invalid access key or secret key
-   * @throws IOException                  upon signature calculation failure
-   * @throws NoSuchAlgorithmException     upon requested algorithm was not found during signature calculation
-   * @throws InvalidExpiresRangeException upon input expires is out of range
+   * @throws InvalidBucketNameException
+   *           upon an invalid bucket name
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws IOException
+   *           upon signature calculation failure
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InvalidExpiresRangeException
+   *           upon input expires is out of range
```

Comment:
IMO you don't need to format these javadoc.  If that's the standard, you could send a separate patch formatting every javadoc all over.